ttnn/cpp/ttnn/operations/data_movement/reshape_view/reshape_pybind.cpp

```diff
@@ -22,22 +24,45 @@ void bind_reshape_view(pybind11::module& module, const data_movement_operation_t
         operation,
         doc,
         ttnn::pybind_overload_t{
-            [](const data_movement_operation_t& self, const ttnn::Tensor& input_tensor, const ttnn::Shape& shape)
-                -> ttnn::Tensor { return self(input_tensor, shape); },
+            [](const data_movement_operation_t& self,
+               const ttnn::Tensor& input_tensor,
+               const ttnn::Shape& shape,
+               const std::optional<MemoryConfig> &memory_config,
+                const uint8_t queue_id,
+                const std::optional<PadValue> &pad_value
+               ) -> ttnn::Tensor {
+                return self(input_tensor, shape);
+            },
             py::arg("input_tensor"),
             py::arg("shape"),
-        },
+            py::kw_only(),
+            py::arg("memory_config") = std::nullopt,
+            py::arg("queue_id") = 0,
+            py::arg("pad_value") = std::nullopt
+            },
         ttnn::pybind_overload_t{
             [](const data_movement_operation_t& self,
                const ttnn::Tensor& input_tensor,
-               const ttnn::SmallVector<int32_t> shape) -> ttnn::Tensor { return self(input_tensor, shape); },
+               const ttnn::SmallVector<int32_t> shape,
+               const std::optional<MemoryConfig> &memory_config,
+               const uint8_t queue_id,
+               const std::optional<PadValue> &pad_value
+               ) -> ttnn::Tensor {
+                return self(input_tensor, shape);
+            },
             py::arg("input_tensor"),
             py::arg("shape"),
-        });
+            py::kw_only(),
+            py::arg("memory_config") = std::nullopt,
+            py::arg("queue_id") = 0,
+            py::arg("pad_value") = std::nullopt
```

Comment:
how do you choose between 0 and nullptr here?

Reply:
I set the default directly when it is an integer and there is no type ambiguity. In the case of pad_value it actually accepts both float or int and interprets it based on the datatype of the tensor. So if it is a bfloat16 tensor, the default is 0.0 but in a uint32 tensor the default is integer 0. In the case of memory_config I read in the configuration of the source tensor and match it to the output tensor if it is not specified.